python/housinginsights/ingestion/LoadData.py

```diff
@@ -286,8 +289,267 @@ def rebuild(self):
 
             processed_data_ids.append(manifest_row['unique_data_id'])
 
+        # build Zone Facts table
+        self._create_zone_facts_table()
+
         return processed_data_ids
 
+    def _create_zone_facts_table(self):
+        """
+        Creates the zone_facts table which is used as a master table for API
+        endpoints. The purpose is to avoid recalculating fields adhoc and
+        performing client-side reconfiguration of data into display fields.
+        This is all now done in the backend whenever new data is loaded.
+
+        :return: a immutable dict that contains the tables structure of the
+        table and sqlAlchemy metadata object used.
+        """
+        # drop zone_facts table if already in db
+        if 'zone_facts' in self.engine.table_names():
+            with self.engine.connect() as conn:
+                conn.execute('DROP TABLE zone_facts;')
+
+        # create empty zone_facts table
+        metadata = MetaData(bind=self.engine)
+        zone_facts = Table('zone_facts', metadata,
+                           Column('zone', String, primary_key=True),
+                           Column('poverty_rate', Numeric),
+                           Column('fraction_black', Numeric),
+                           Column('income_per_capita', Numeric),
+                           Column('labor_participation', Numeric),
+                           Column('fraction_foreign', Numeric),
+                           Column('fraction_single_mothers', Numeric),
+                           Column('acs_lower_rent_quartile', Numeric),
+                           Column('acs_median_rent', Numeric),
+                           Column('acs_upper_rent_quartile', Numeric))
+
+        # add to db
+        metadata.create_all(tables=[zone_facts])
+
+        # populate table with calculated fields values
+        self._populate_zone_facts_table()
+
+        return metadata.tables
+
+    def _populate_zone_facts_table(self):
+        """
+        Populates the zone_facts table with the calculated fields data and
+        acs rent data fields from census.
+
+        :return: list of SQLAlchemy query result objects from inserting the
+        calculated fields data into the zone_facts table.
+        """
+        census_fields = [
```

Comment:
For reference for future developers, should add a comment here describing the purpose of this list. Contains the list of field names to be included in the zone_facts table. Names should correspond to either a) a column name in the 'census' table or b) a calculated field as defined in the self._census_with_weighting function.